components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/IdentityProviderManagementServiceTest.java

```diff
@@ -0,0 +1,904 @@
+        prc.setRoleMappings(new RoleMapping[]{rm1, rm2});
+        idp1.setPermissionAndRoleConfig(prc);
+
+        FederatedAuthenticatorConfig fac = new FederatedAuthenticatorConfig();
+        fac.setDisplayName("DisplayName1");
+        fac.setName("Name");
+        fac.setEnabled(true);
+        Property property1 = new Property();
+        property1.setName("Property1");
+        property1.setValue("value1");
+        property1.setConfidential(false);
+        Property property2 = new Property();
+        property2.setName("Property2");
+        property2.setValue("value2");
+        property2.setConfidential(true);
+        fac.setProperties(new Property[] {property1, property2});
+        idp1.setFederatedAuthenticatorConfigs(new FederatedAuthenticatorConfig[] {fac});
+
+        ProvisioningConnectorConfig pcc1 = new ProvisioningConnectorConfig();
+        pcc1.setName("ProvisiningConfig1");
+        pcc1.setProvisioningProperties(new Property[] {property1});
+        ProvisioningConnectorConfig pcc2 = new ProvisioningConnectorConfig();
+        pcc2.setName("ProvisiningConfig2");
+        pcc2.setProvisioningProperties(new Property[] {property2});
+        pcc2.setEnabled(true);
+        pcc2.setBlocking(true);
+        idp1.setProvisioningConnectorConfigs(new ProvisioningConnectorConfig[] {pcc1, pcc2});
+
+        ClaimConfig claimConfig = new ClaimConfig();
+        claimConfig.setLocalClaimDialect(false);
+        claimConfig.setRoleClaimURI("Country");
+        claimConfig.setUserClaimURI("Country");
+        ClaimMapping cm = ClaimMapping.build("http://wso2.org/claims/country", "Country", "", true);
+        claimConfig.setClaimMappings(new ClaimMapping[]{cm});
+        Claim remoteClaim = new Claim();
+        remoteClaim.setClaimId(0);
+        remoteClaim.setClaimUri("Country");
+        claimConfig.setIdpClaims(new Claim[]{remoteClaim});
+        idp1.setClaimConfig(claimConfig);
+
+        //Initialize Test Identity Provider 2
+        IdentityProvider idp2 = new IdentityProvider();
+        idp2.setIdentityProviderName("testIdP2");
+
+        ClaimConfig claimConfig2 = new ClaimConfig();
+        claimConfig2.setLocalClaimDialect(true);
+        claimConfig2.setRoleClaimURI("http://wso2.org/claims/role");
+        claimConfig2.setUserClaimURI("http://wso2.org/claims/fullname");
+        ClaimMapping cm2 = new ClaimMapping();
+        Claim localClaim2 = new Claim();
+        localClaim2.setClaimId(0);
+        localClaim2.setClaimUri("http://wso2.org/claims/fullname");
+        cm2.setLocalClaim(localClaim2);
+        claimConfig2.setClaimMappings(new ClaimMapping[]{cm2});
+        idp2.setClaimConfig(claimConfig2);
+
+        //Initialize Test Identity Provider 3
+        IdentityProvider idp3 = new IdentityProvider();
+        idp3.setIdentityProviderName("testIdP3");
+
+        return new Object[][]{
+                {idp1},
+                {idp2},
+                {idp3},
```

Comment:
can you have a comment describing each idp and how it would be different from the rest of the IdPs so that it was included as separate entry?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/3493/commits/09fbd1468a2abedaaf4c5f79daa567709fcbda99